src/routine.cpp

```diff
@@ -24,6 +24,7 @@ namespace clblast {
 // The constructor does all heavy work, errors are returned as exceptions
 Routine::Routine(Queue &queue, EventPointer event, const std::string &name,
                  const std::vector<std::string> &routines, const Precision precision,
+                 std::initializer_list<const char *> source_pct,
```

Comment:
I meant the third routine mode, not kernel mode. There are two enums.

Initially I wanted to provide a way for a plugin to override only a part of the host routine code (i. e. the built-in routine code will call built-in PadCopyTranspose, and then the custom routine code will call the custom main kernel), but this is too inflexible.